File: javascript/3. Longest Substring Without Repeating Characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function (s) {
	max = 0;
	substring = [];
	[...s].forEach((c) => {
		if (substring.includes(c)) {
			max = Math.max(max, substring.length);
			idx = substring.indexOf(c);

			substring.splice(0, idx + 1);
		}

		substring.push(c);
	});
	max = Math.max(max, substring.length);

	return max;
};

console.log(lengthOfLongestSubstring('dvdf'));

File: Blind Golang/String/3. Longest Substring Without Repeating Characters.go
package string

func lengthOfLongestSubstring(s string) int {

	if len(s) == 0 {
		return 0
	}

	max := 1
	curSub := s[:0]
	for i := 1; i < len(s); i++ {
		isExist := false
		for j := 0; j < len(curSub); j++ {
			if s[i] == curSub[j] {
				curSub = curSub[j+1:]
				curSub += string(s[i])
				isExist = true
				break
			}
		}

		if !isExist {
			curSub += string(s[i])
			if len(curSub) > max {
				max = len(curSub)
			}
		}
	}

	return max
}
